src/server/game/Globals/ObjectMgr.cpp

```diff
@@ -9225,20 +9225,32 @@ std::string const& ObjectMgr::GetScriptName(uint32 id) const
     return (id < _scriptNamesStore.size()) ? _scriptNamesStore[id] : empty;
 }
 
-uint32 ObjectMgr::GetScriptId(std::string const& name)
+uint32 ObjectMgr::GetScriptIdOrAdd(std::string const& name)
 {
     // use binary search to find the script name in the sorted vector
     // assume "" is the first element
     if (name.empty())
         return 0;
 
-    ScriptNameContainer::const_iterator itr = std::lower_bound(_scriptNamesStore.begin(), _scriptNamesStore.end(), name);
+    ScriptNameContainer::const_iterator itr = std::find(_scriptNamesStore.begin(), _scriptNamesStore.end(), name);
     if (itr == _scriptNamesStore.end() || *itr != name)
-        return 0;
+    {
+        _scriptNamesStore.push_back(name);
```

Comment:
Is this thread safe ?

Reply:
No its probably not. If the core uses multiple threads to spawn entities it would probably create a issue here. Any ideas about how to make it thread safe?

Reply:
> 
> 
> No its probably not. If the core uses multiple threads to spawn entities it would probably create a issue here. Any ideas about how to make it thread safe?

Wrong. The only time this function can possibly be called when the scriptname isnt known is during reload triggered by gm/console command running in main thread (its safe)

Reply:
For fucks sake Shauren xD you're right again, just checked, its not called on spawn or smth, only when you're reloading or loading at core start. Well yeah you already said the other thing.